src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1596,7 +1638,9 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageEndSession(
                          current_session_id,
                          packet.protocol_version(),
                          service_type,
-                         rejectedParams);
+                         rejectedParams,
+                         "Refused to end session for service of type " +
```

Comment:
`OnSessionEndedCallback` has some descriptive logs in it with its error returns, do you think this reason param is enough for developers, or should some message be passed up from `OnSessionEndedCallback` or even deeper?

Reply:
Implemented getting the failure reason from `OnSessionEndedCallback` in https://github.com/smartdevicelink/sdl_core/pull/3462/commits/4c55f0b4aba9f5887d71dbcc36423967e550d736